macos/ds_store.ksy

```diff
@@ -0,0 +1,195 @@
+meta:
+  id: ds_store
+  title: macOS '.DS_Store' format
+  license: CC-BY-SA-4.0
+  ks-version: 0.8
+  endian: be
+doc: |
+  Apple macOS '.DS_Store' file format.
+doc-ref: |
+  https://en.wikipedia.org/wiki/.DS_Store
+  https://metacpan.org/pod/distribution/Mac-Finder-DSStore/DSStoreFormat.pod
+  https://0day.work/parsing-the-ds_store-file-format
+seq:
+  - id: alignment_header
+    contents: [0x00, 0x00, 0x00, 0x01]
+  - id: buddy_allocator_header
+    type: buddy_allocator_header
+instances:
+  buddy_allocator_body:
+    pos: buddy_allocator_header.ofs_bookkeeping_info_block + 4
+    size: buddy_allocator_header.len_bookkeeping_info_block
+    type: buddy_allocator_body
+types:
+  buddy_allocator_header:
+    seq:
+      - id: magic
+        contents: [0x42, 0x75, 0x64, 0x31]
+        doc: Magic number 'Bud1'
+      - id: ofs_bookkeeping_info_block
+        type: u4
+      - id: len_bookkeeping_info_block
+        type: u4
+      - id: copy_ofs_bookkeeping_info_block
+        type: u4
+        doc: Needs to match 'offset_bookkeeping_info_block'
+      - size: 16
+        doc: Unused block
+  buddy_allocator_body:
+    seq:
+      - id: num_blocks
+        type: u4
+        doc: Number of blocks in the allocated-blocks list
+      - size: 4
+        doc: Unknown field
+      - id: block_addresses
+        type: u4
+        repeat: expr
+        repeat-expr: 256
+        doc: Addresses of the different blocks
+      - id: num_directories
+        type: u4
+        doc: Indicates the number of directory entries
+      - id: directory_entries
+        type: directory_entry
+        repeat: expr
+        repeat-expr: num_directories
+        doc: Each directory is an independent B-tree
+      - id: free_lists
+        type: free_list
+        repeat: expr
+        repeat-expr: 32
+    instances:
+      directories:
+        io: _root._io
+        type: master_block_ref(_index)
+        repeat: expr
+        repeat-expr: num_directories
+        doc: Master blocks of the different B-trees
+  directory_entry:
+    seq:
+      - id: len_name
+        type: u1
+      - id: name
+        size: len_name
+        type: str
+        encoding: UTF-8
+      - id: block_id
+        type: u4
+  free_list:
+    seq:
+      - id: counter
+        type: u4
+      - id: offsets
+        type: u4
+        repeat: expr
+        repeat-expr: counter
+  master_block_ref:
+    params:
+      - id: idx
+        type: u8
+    instances:
+      master_block:
+        pos: (_parent.block_addresses[_parent.directory_entries[idx].block_id] >> 0x05 << 0x05) + 4
```

Comment:
1. `& ~0x1f` ?
2. snould bit-sized types be used?

Reply:
1. Yes you're right. It's cleaner and shorter :+1:
2. I'm not sure to see how you would use bit-sized types within this case. I use bitwise operations to calculate the position and the size of some blocks which have their own type.